SonarQube.Client/Api/V7_20/GetIssuesRequestWrapper.cs

```diff
@@ -60,29 +59,29 @@ public async Task<SonarQubeIssue[]> InvokeAsync(HttpClient httpClient, Cancellat
             ResetInnerRequest();
             innerRequest.Types = "CODE_SMELL";
             var codeSmells = await innerRequest.InvokeAsync(httpClient, token);
-            WarnForApiLimit(codeSmells);
+            WarnForApiLimit(codeSmells, innerRequest);
 
             ResetInnerRequest();
             innerRequest.Types = "BUG";
             var bugs = await innerRequest.InvokeAsync(httpClient, token);
-            WarnForApiLimit(bugs);
+            WarnForApiLimit(bugs, innerRequest);
 
             ResetInnerRequest();
             innerRequest.Types = "VULNERABILITY";
             var vulnerabilities = await innerRequest.InvokeAsync(httpClient, token);
-            WarnForApiLimit(vulnerabilities);
+            WarnForApiLimit(vulnerabilities, innerRequest);
 
             return codeSmells
                 .Concat(bugs)
                 .Concat(vulnerabilities)
                 .ToArray();
         }
 
-        private void WarnForApiLimit(SonarQubeIssue[] issues)
+        private void WarnForApiLimit(SonarQubeIssue[] issues, GetIssuesRequest request)
         {
-            if (issues.Length == 10000)
+            if (issues.Length == request.ItemsLimit)
             {
-                Logger.Warning($"The SonarQube maximum API response limit reached. Some issues might not be suppressed.");
+                Logger.Warning($"The SonarQube maximum API response limit reached. Some issues might not be suppressed, suppressing the first {request.ItemsLimit} issues.");
```

Comment:
The sentence is a bit clumsy and I don't think it's grammatically correct (in fact, the original sentence wasn't grammatically correct either). It might be better to not refer to SonarQube specifically since SonarCloud users will see the same message. Here a couple of alternatives:

>Sonar web API response limit reached ({request.ItemsLimit} items). Some issues might not be suppressed.

or 

>Sonar web maximum API response limit reached. Only the first {request.ItemsLimit} issues will be suppressed.